src/EnergyPlus/Material.cc

```diff
@@ -2756,6 +2758,107 @@ void GetMaterialData(EnergyPlusData &state, bool &ErrorsFound) // set to true if
     for (auto *m : state.dataMaterial->Material) {
         m->phaseChange = HysteresisPhaseChange::HysteresisPhaseChange::factory(state, m->Name);
     }
+
+    GetVariableAbsorptanceInput(state, ErrorsFound); // Read variable thermal and solar absorptance add-on data
+}
+
+void GetVariableAbsorptanceInput(EnergyPlusData &state, bool &errorsFound)
+{
+    int IOStat; // IO Status when calling get input subroutine
+    int numAlphas;
+    int numNumbers;
+    Array1D_string alphas(7);   // character string data
+    Array1D<Real64> numbers(1); // numeric data
+    std::string_view cCurrentModuleObject{"MaterialProperty:VariableAbsorptance"};
+    int numVariAbs = state.dataInputProcessing->inputProcessor->getNumObjectsFound(state, cCurrentModuleObject);
+    state.dataHeatBal->AnyVariableAbsorptance = (numVariAbs > 0);
+    for (int i = 1; i <= numVariAbs; ++i) {
+        // Call Input Get routine to retrieve material data
+        state.dataInputProcessing->inputProcessor->getObjectItem(state,
+                                                                 cCurrentModuleObject,
+                                                                 i,
+                                                                 alphas,
+                                                                 numAlphas,
+                                                                 numbers,
+                                                                 numNumbers,
+                                                                 IOStat,
+                                                                 state.dataIPShortCut->lNumericFieldBlanks,
+                                                                 state.dataIPShortCut->lAlphaFieldBlanks,
+                                                                 state.dataIPShortCut->cAlphaFieldNames,
+                                                                 state.dataIPShortCut->cNumericFieldNames);
+
+        // Load the material derived type from the input data.
+        int MaterNum = UtilityRoutines::FindItemInPtrList(alphas(2), state.dataMaterial->Material);
+        if (MaterNum == 0) {
+            ShowSevereError(state,
+                            format("{}: invalid {} entered={}, must match to a valid Material name.",
+                                   cCurrentModuleObject,
+                                   state.dataIPShortCut->cAlphaFieldNames(2),
+                                   alphas(2)));
+            errorsFound = true;
+            return;
+        }
+        auto *thisMaterial = state.dataMaterial->Material(MaterNum);
```

Comment:
Auto with pointer feels a little vague to me.  I'd like to minimize this pattern in the future.  I think it would be easy enough to be explicit about the type here, it's not like it's a long iterator typename.  This is not critical for merge today.

Reply:
What? Why? This isn't an iterator, it's no different than `auto &`. The reason we are using `*` here instead of `&` is that we changed the Material array to be an array of `Material` pointers rather than `Material` objects so that we can take advantage of inheritance (we're not doing it yet, but it is coming).

Reply:
My issue with it is because of what auto means when you are assigning to a pointer object.  In this case, both of these are the same thing:

```
auto * thisMaterial = state.dataMaterial->Material(MaterNum);
auto thisMaterial = state.dataMaterial->Material(MaterNum);
```

They both result in thisMaterial being a pointer to a material.  I'm not upset about there being a pointer, I just dislike that auto sneaks in the pointer automatically, meaning...you don't need the * anymore.  Whereas if you use the typename, you get:

```
MaterialProperties * thisMaterial = state.dataMaterial->Material(MaterNum);
MaterialProperties thisMaterial = state.dataMaterial->Material(MaterNum);  << COMPILER ERROR
```

It's a small change to make things super clear.  I just don't like that `auto *` can produce either an instance or a pointer type, and will ignore the * if it's already a pointer type.  Like...what if you actually wanted a pointer to a pointer and didn't want the compiler to ignore the *?

This is extremely far from a hill I would die on, so feel free to > /dev/null as needed.

Demo: https://godbolt.org/z/jhjjPb4ob

Reply:
I would think about consenting to this if we were not using `auto &` so extensively. I still wouldn't consent to it, but I would think about it. 

We need to severely limit the ways in which we use "plain" `auto`. A good number of current uses of "plain" `auto` should be `auto &` to prevent unnecessary deep copies. If we use `auto thisMaterial = state.dataMaterial->Material(MaterialNum);` then it looks like one of those bad instances which we are trying to hunt down and convert to `auto &`. 

If you're starting to get the feeling that `auto` is a nightmare that is much more trouble than it's worth, then you're onto something.